### PR TITLE
Add `sentry`

### LINKED ISSUE #5440: Investigate Sentry in Electron & Web
Try sentry in `rust/js` to see how it renders stacktrace on Sentry

### PR DESCRIPTION
**Pull request by TimeEngineer**
closes #5440 
It looks like that in `sentry`:
image

### CODE REVIEW

File: bindings/electron/src/lib.rs [hunk 1]
```diff
@@ -10,6 +12,22 @@ lazy_static::lazy_static! {
...
+        SENTRY.get_or_init(|| {
+            sentry::init((
+                sentry_url,
+                ClientOptions {
+                    release: sentry::release_name!(),
+                    ..Default::default()
+                },
+            ))
+        });
+    }
```
Comment (FirelightFlagboy) [inline bindings/electron/src/lib.rs:28]: Should we add a log indicating that sentry was or wasn't configured ?

File: bindings/electron/src/meths.rs
Comment (FirelightFlagboy) [inline bindings/electron/src/meths.rs:1]: Can't the electron just call `init_sentry` a single time ?

File: bindings/web/src/lib.rs [hunk 2]
```diff
@@ -2,12 +2,30 @@
...
 #[allow(unused_imports)]
 use wasm_bindgen::prelude::*;
 
 mod meths;
 pub use meths::*;
 
+static SENTRY: OnceCell<ClientInitGuard> = OnceCell::new();
+
+fn init_sentry() {
+    if let Ok(sentry_url) = std::env::var("SENTRY_URL") {
```
Comment (FirelightFlagboy) [inline bindings/web/src/lib.rs:16]: Env variable in web ? how does that work ?

File: bindings/web/src/meths.rs
Comment (FirelightFlagboy) [inline bindings/web/src/meths.rs:1]: Like `electron` can't the web warmup the binding by calling `init_sentry` a single time ?

**Review by FirelightFlagboy — COMMENTED** [review]
So we choose to have 2 sentry-client running at the same time ?

**Comment by TimeEngineer** [comment]
> So we choose to have 2 sentry-client running at the same time ? (on javascript & rust)

We haven't decided yet

File: bindings/electron/src/meths.rs
Comment (TimeEngineer) [reply bindings/electron/src/meths.rs:1]: The problem is that we don't have a main thread accessible in `rust`

File: bindings/electron/src/lib.rs (hunk 1 above)
Comment (TimeEngineer) [reply bindings/electron/src/lib.rs:28]: @touilleMan

File: bindings/web/src/lib.rs (hunk 2 above)
Comment (FirelightFlagboy) [reply bindings/web/src/lib.rs]: Still, how does that work ?

File: bindings/electron/src/meths.rs
Comment (TimeEngineer) [reply bindings/electron/src/meths.rs:1]: See: https://docs.sentry.io/platforms/rust/#async-main-function

**Comment by mmmarcos** [comment]
This work is included in #7814

**Pull request closed without merging**